graphql-kotlin-schema-generator/src/test/kotlin/com/expediagroup/graphql/test/integration/InterfaceOfInterfaceTest.kt

```diff
@@ -61,5 +77,12 @@ class InterfaceOfInterfaceTest {
 
     class InterfaceOfInterfaceQuery {
         fun getClass() = MyClass(id = "1", name = "fooBar")
+        fun getIgnoredClass(): SecondLevel = IgnoredClass(id = "2", name = "baz")
```

Comment:
this seems weird... so we return an interface without actual implementation?

Reply:
Yes, this can be helpful for cases where maybe you have some shared interfaces.

```kotlin
interface MySharedInnerType {
  val value: String
}

interface MySharedType {
  val id: String
  val value: MySharedInnerType
}
```

So then in the code you only want the services to ever use and expose the `MySharedType` but if the API developers wanted different implementations they could create that logic but clients would still only ever see interfaces.

Reply:
still seems "wrong" to expose only interfaces without implementations, wonder if something like this causes issues for the clients (it definitely would for our own client code generation)

Reply:
I can close this PR if needed. This just verifies behaviour that is possible with the generator. Whether you do it or not is up to your schema design

Reply:
I understand that this is a current behavior. My questions were more about asking whether this should be allowed or not. I don't see anything in the spec that explicitly says interface must have any implementations....

Reply:
No, and when you get into the schema stitching/federation world, we may have a published schema that is only interfaces and maybe only in the final merged schemas is there implementations but even then the fields just might only be interfaces.

For clients it doesn't matter when writing queries if the types (implementations) don't expose any additional fields

Reply:
If we want to change the behaviour in the future this test will verify we have done it correctly by failing